### DISCUSSION TITLE
Can we expose a hook when signing the application so that users can complete the signing of the bin file by themselves?

### DISCUSSION THREAD

**Discussion by muwoo** [tauri-apps/tauri#9900] [Q&A] [answered]
In [electron-builder](https://www.electron.build/configuration/win), we can complete the signature of the required files by ourselves in this way:
```json
{
  "build": {
    "win": {
      "sign":  "./sign.js"
    }
  }
}
```

The `sign` field here can be a Promise function (or path to file or module id). When `electron-builder` is building, it will dynamically pass the file path that needs to be signed to this function, so that the logic of signing the file can be customized within this function.

Can Tauri also expose a similar hook such as `signCommand`? Then when executing the signature, it is judged that if there is this parameter, Tauri's built-in signature logic will not be used, and `signCommand` will be called directly and the file that needs to be signed will be passed in.

We really need the custom signature function, and we look forward to your reply.

**Comment by FabianLars** [comment]
This was implemented a few days ago but iirc for v2 only (in the next beta release?)

**Reply by muwoo** [reply]
Thank you for your reply. May I ask if this is the PR?

 https://github.com/tauri-apps/tauri/pull/9731

**Reply by FabianLars** [reply, marked as answer]
https://github.com/tauri-apps/tauri/pull/9865

**Reply by muwoo** [reply]
Well, thanks, can we migrate this feature to the v1.x branch? I really plan on making a PullRequest
